server/lib/src/idm/oauth2.rs

```diff
@@ -248,24 +248,32 @@ impl AuthorisePermitSuccess {
     pub fn build_redirect_uri(&self) -> Url {
         let mut redirect_uri = self.redirect_uri.clone();
 
-        // Always clear query and fragment, regardless of the response mode
-        redirect_uri.set_query(None);
+        // Always clear the fragment per RFC
         redirect_uri.set_fragment(None);
 
-        // We can't set query pairs on fragments, only query.
-        let mut uri_builder = url::form_urlencoded::Serializer::new(String::new());
-
-        uri_builder.append_pair("code", &self.code);
-
-        if let Some(state) = self.state.as_ref() {
-            uri_builder.append_pair("state", state);
-        };
+        match self.response_mode {
+            ResponseMode::Query => {
+                redirect_uri
+                    .query_pairs_mut()
+                    .append_pair("code", &self.code);
```

Comment:
Will this conflict if the query was sent with a `?code=...` query parameter?

Reply:
We stomp it in that case, and they can get in the bin.

Reply:
Just making sure we don't return `? code=somecode&code=someothercode`